### ISSUE TITLE
Classification on large *.ply data

### ISSUE THREAD

**Issue by tsly123** [yangyanli/PointCNN#138]
Hi,
Thank you for sharing your work. My 3D face dataset contains large `ply` files. For example:
```
ply
format ascii 1.0
element vertex 166368
property float x
property float y
property float z
element face 156797
property list uchar int vertex_indices
end_header
-50.8063 31.0753 83.1526
...
3 37583 37611 37610 
...
```
And i should do the classification basing on these ply files but I have no idea how to implement it using PointCNN. What should i do?

Thank you.
tsly

**Comment by TSchattschneider** [comment]
You could implement a function similar to [`load_cls()`](https://github.com/yangyanli/PointCNN/blob/master/data_utils.py#L121) and use the `plyfile` python library instead of `h5py` to load in your point data from your PLY. You would for example similarly put all the points into a list (called `points` in the referenced function) and then use numpy's concatenation function to create a single numpy array.
You could then return that array along with a corresponding label array and use this function to load in your data in customized versions of the training/testing scripts for classification.

**Comment by tsly123** [comment]
Thank you @TSchattschneider ,
So far, I've tried to run the mnist, cifar examples and now I am struggling to extract the feature from `FC` and also `xconv layers`. Do you know how to extract feature from PointCNN pretrained network? 
Thank you.
tsly

**Comment by burui11087** [comment]
@tsly123 
Hi, please use this command `docker pull tensorflow/tensorflow:1.6.0-devel-gpu-py3` and install nvidia-docker environment.

**Comment by tsly123** [comment]
Hi,
I apologize to bother you again. I am struggling to extract feature from `xconv layer before fc0` (and hopefully `fc_mean` layer) in order of `train_paths` not shuffling the dataset. I tried to code and faced many errors. Below is one of my trials.
```
with tf.Graph().as_default():
    with tf.Session() as sess:
        new_saver = tf.train.import_meta_graph(meta_path)
        new_saver.restore(sess, ckpt_path)
        input_fc0 = tf.get_default_graph().get_tensor_by_name('xconv_6_fts_list_concat:0')
        input_fc0 = tf.nn.l2_normalize(input_fc0,2)
        input_fc0_size = input_fc0.get_shape()[-1]
        images_placeholder = tf.get_default_graph().get_tensor_by_name("data_train:0")
        phase_train_placeholder = tf.get_default_graph().get_tensor_by_name("is_training:0")
        handle = tf.get_default_graph().get_tensor_by_name("handle:0")
        
        emb_array = np.zeros((len(train_paths), 128, input_fc0_size))
        array = np.zeros((26929, 4096, 6))  #26929 = nb of samples
        for idx, arr_path in enumerate(train_paths):
            array_file = np.loadtxt(arr_path, delimiter=' ')
            array = np.zeros((array_file.shape[0], 6))  
            array[:,:3] = array_file
            array[:,-1] = 1
        
        dataset = tf.data.Dataset.from_tensor_slices((array))
        iterator = dataset.make_one_shot_iterator()
        handle_feed = sess.run(iterator.string_handle())

        for i in range(0, len(train_paths)):
            inputs = iterator.get_next()
            feed_dict = { images_placeholder:inputs, handle: handle_feed, phase_train_placeholder:False }
            arr = sess.run(input_fc0, feed_dict=feed_dict)

            emb_array[i,:] = arr
        
        emb_array2 = np.mean(emb_array, axis = 1).reshape(len(train_paths), input_fc0_size)    
``` 
and it raise `couldn't create tensor over 2.0Gb`. I've also tried to break the data into batches but it keep continuing raise difference errors, such as `couldn't feed a sample shape (4096, 6) to 'data_train:0' which has shape (26926, 4096, 6)`.
Could you give me and example of extracting feature from xconv layer? I'm new to tensorflow framework. 

Thank you.
tsly

**Comment by burui11087** [comment]
hi @tsly123 

you can return the fts_X variable in  pointcnn.py such as https://github.com/yangyanli/PointCNN/blob/56182cded98f00b280e3ee16fc46004dc165ed4b/pointcnn.py#L42
